sdk/storage/storage-blob/samples/browserSamples/largeFileUploads.html

```diff
@@ -0,0 +1,109 @@
+<!--
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT Licence.
+
+  Here we demonstrate how to inject new SAS for an ongoing uploading. 
+  SAS might get expired before a large upload finishes, for this scenario, we want to request a new SAS token during the upload instead of starting a new upload.
+  
+  In this sample, we give a SAS injection sample for browsers like Chrome which supports await/async.
+  Sample code imports Azure Storage Blob SDK with JavaScript bundle.
+  Before executing the sample: 
+  - Make sure storage account has been set up CORS setting properly
+  - Implement method getNewSasForBlob 
+  - Update url in upload() method
+
+
+  
+  Refer to other Javascript samples(https://github.com/Azure/azure-sdk-for-js/tree/feature/storage/sdk/storage/storage-blob/samples/javascript) 
+  and replace code in the below script tag to try out other features.
+-->
+
+<html>
+  <body>
+    <form><input type="file" id="file" /></form>
+    <button id="upload" onclick="upload()">Upload</button>
+
+    <!-- Import browser bundle `azure-storage-blob.js` or `azure-storage-blob.min.js`-->
+    <script src="./azure-storage-blob.js"></script>
+
+    <script>
+      // azblob is default exported entry when importing Azure Storage Blob SDK by <script src="azure-storage.blob.js">
+
+      // SasStore is a class to cache SAS for blobs
+      class SasStore {
+        constructor() {
+          this.sasCache = {};
+        }
+
+        // Get a valid SAS for blob
+        async getValidSASForBlob(blobURL) {
+          if (this.sasCache[blobURL] && this.isSasStillValidInNext2Mins(this.sasCache[blobURL])) {
+            return this.sasCache[blobURL];
+          } else {
+            return (this.sasCache[blobURL] = await this.getNewSasForBlob(blobURL));
+          }
+        }
+
+        // Return true if "se" section in SAS is still valid in next 2 mins
+        isSasStillValidInNext2Mins(sas) {
+          const expiryStringInSas = new URL(`http://hostname${sas}`).searchParams.get("se");
+          return new Date(expiryStringInSas) - new Date() >= 2 * 60 * 1000;
+        }
+
+        // Get a new SAS for blob, we assume a SAS starts with a "?"
+        async getNewSasForBlob(blobURL) {
+          // TODO: You need to implement this
+          return "?newSAS";
```

Comment:
Should we provide a sample code here ?

Reply:
SDK doesn't allow generate SAS in browser environment, so they should get a new SAS from their service backend.